server/src/com/cloud/vm/snapshot/VMSnapshotManagerImpl.java

```diff
@@ -610,7 +615,11 @@ public UserVm revertToSnapshot(Long vmSnapshotId) throws InsufficientCapacityExc
             VmWorkJobVO placeHolder = null;
             placeHolder = createPlaceHolderWork(vmSnapshotVo.getVmId());
             try {
-                return orchestrateRevertToVMSnapshot(vmSnapshotId);
+                UserVm revertedVM = orchestrateRevertToVMSnapshot(vmSnapshotId);
+
+                updateUserVmServiceOffering(revertedVM, vmSnapshotVo);
```

Comment:
Better to put updateUserVmServiceOffering() call inside orchestrateRevertToVMSnapshot() as orchestrate is called from multiple places.